### PR TITLE
File Manager - Provide context menu in details view and display parent folder in files view

### PR DESCRIPTION
**Pull request by maxerokh**
Key functionalities:
- Context menu in details view
- Display parent folder in details and thumbnails view
- Process click by folder in details view. After click, end user will navigate to folder's content.
- Added thumbnail column in details view

### CODE REVIEW

File: js/ui/file_manager/ui.file_manager.commands.js [hunk 1]
```diff
@@ -0,0 +1,90 @@
+let tryCreateFolder = function(fileManager) {
```
Comment (jsdmitry) [inline js/ui/file_manager/ui.file_manager.commands.js:1]: if tryCreateFolder isn't changing then use `const` instead of `let`

Comment (vladkovl) [reply js/ui/file_manager/ui.file_manager.commands.js:1]: We plan to refactor this code during the implementation of the next feature - toolbar support

File: testing/tests/DevExpress.ui.widgets/fileManagerParts/contextMenu.tests.js [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+    afterEach: () => {
+        this.clock.tick(5000);
+
+        this.clock.restore();
+        fx.off = false;
+    }
+};
+
+let getRowActionButtonInDetailsView = function($element, index) {
```
Comment (jsdmitry) [inline testing/tests/DevExpress.ui.widgets/fileManagerParts/contextMenu.tests.js]: Remove `function` and use `const`

Comment (jsdmitry) [reply testing/tests/DevExpress.ui.widgets/fileManagerParts/contextMenu.tests.js]: And can write this code in the one line

File: testing/tests/DevExpress.ui.widgets/fileManagerParts/contextMenu.tests.js [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+        getRowNameCellInDetailsView(this.$element, 2).trigger("dxcontextmenu");
+        assert.notOk($row1.hasClass("dx-selection"));
+        assert.ok($row2.hasClass("dx-selection"));
+        assert.ok(getContextMenuItems().length > 0);
+    });
+
+    test('right click by row and click by select check box', assert => {
+        getSelectCheckBoxInDetailsView(this.$element, 1).trigger("dxclick");
+
+        let $row1 = getRowInDetailsView(this.$element, 1);
```
Comment (jsdmitry) [inline testing/tests/DevExpress.ui.widgets/fileManagerParts/contextMenu.tests.js]: Use `const`  !

File: testing/tests/DevExpress.ui.widgets/fileManagerParts/contextMenu.tests.js (hunk 2 above)
Comment (vladkovl) [reply testing/tests/DevExpress.ui.widgets/fileManagerParts/contextMenu.tests.js]: Done

File: testing/tests/DevExpress.ui.widgets/fileManagerParts/contextMenu.tests.js (hunk 3 above)
Comment (vladkovl) [reply testing/tests/DevExpress.ui.widgets/fileManagerParts/contextMenu.tests.js]: Done

**Review by jsdmitry — DISMISSED** [review]

File: js/ui/file_manager/ui.file_manager.item_list.details.js [hunk 4]
```diff
@@ -1,12 +1,17 @@
 import $ from "../../core/renderer";
+import iconUtils from "../../core/utils/icon";
```
Comment (DokaRus) [inline js/ui/file_manager/ui.file_manager.item_list.details.js]:
```suggestion
import { getImageContainer } from "../../core/utils/icon";
```

File: js/ui/file_manager/ui.file_manager.item_list.details.js [hunk 5]
```diff
@@ -18,49 +23,71 @@ class FileManagerDetailsItemList extends FileManagerItemListBase {
...
+    _createContextMenuItems(fileItem) {
+        const isSingleRowSelected = this._filesView.getSelectedRowKeys().length <= 1;
+        return FileManagerFileCommands
+            .filter(c => {
+                if(c.displayInToolbarOnly) {
+                    return false;
+                }
+                return !c.isSingleFileItemCommand || c.isSingleFileItemCommand === isSingleRowSelected;
+            })
+            .map(c => ({
```
Comment (DokaRus) [inline js/ui/file_manager/ui.file_manager.item_list.details.js]:
```suggestion
            .map({ text, name } => ({
```

File: js/ui/file_manager/ui.file_manager.item_list.details.js [hunk 6]
```diff
@@ -18,49 +23,71 @@ class FileManagerDetailsItemList extends FileManagerItemListBase {
...
+                name: c.name,
```
Comment (DokaRus) [inline js/ui/file_manager/ui.file_manager.item_list.details.js]:
```suggestion
                name,
```

File: js/ui/file_manager/ui.file_manager.item_list.details.js [hunk 7]
```diff
@@ -18,49 +23,71 @@ class FileManagerDetailsItemList extends FileManagerItemListBase {
...
+                text: c.text,
```
Comment (DokaRus) [inline js/ui/file_manager/ui.file_manager.item_list.details.js]:
```suggestion
                text,
```

File: js/ui/file_manager/ui.file_manager.item_list.details.js [hunk 8]
```diff
@@ -18,49 +23,71 @@ class FileManagerDetailsItemList extends FileManagerItemListBase {
...
+                fileItem: fileItem,
```
Comment (DokaRus) [inline js/ui/file_manager/ui.file_manager.item_list.details.js]:
```suggestion
                fileItem,
```

File: js/ui/file_manager/ui.file_manager.item_list.details.js [hunk 9]
```diff
@@ -71,6 +98,71 @@ class FileManagerDetailsItemList extends FileManagerItemListBase {
...
+        }
+
+        const item = e.row.data;
+        e.items = this._createContextMenuItems(item);
+        this._ensureItemSelected(item);
+    }
+
+    _createThumbnailColumnCell(container, cellInfo) {
+        const thumbnail = this._getItemThumbnail(cellInfo.data);
+        iconUtils.getImageContainer(thumbnail)
```
Comment (DokaRus) [inline js/ui/file_manager/ui.file_manager.item_list.details.js]:
```suggestion
        getImageContainer(thumbnail)
```

File: js/ui/file_manager/ui.file_manager.item_list.js [hunk 10]
```diff
@@ -1,13 +1,45 @@
 import { extend } from "../../core/utils/extend";
+import dblclickEvent from "../../events/double_click";
+import eventUtils from "../../events/utils";
```
Comment (DokaRus) [inline js/ui/file_manager/ui.file_manager.item_list.js]:
```suggestion
import { addNamespace } from "../../events/utils";
```

File: js/ui/file_manager/ui.file_manager.item_list.js [hunk 11]
```diff
@@ -1,13 +1,45 @@
...
 
 const FILE_MANAGER_FILES_VIEW_CLASS = "dx-filemanager-files-view";
+const FILE_MANAGER_ITEM_LIST_ITEM_OPEN_EVENT_NAMESPACE = "dxFileManager_open";
 
 class FileManagerItemListBase extends Widget {
 
     _initMarkup() {
         this.$element().addClass(FILE_MANAGER_FILES_VIEW_CLASS);
+
+        const dblClickEventName = eventUtils.addNamespace(dblclickEvent.name, FILE_MANAGER_ITEM_LIST_ITEM_OPEN_EVENT_NAMESPACE);
```
Comment (DokaRus) [inline js/ui/file_manager/ui.file_manager.item_list.js]:
```suggestion
        const dblClickEventName = addNamespace(dblClickName, FILE_MANAGER_ITEM_LIST_ITEM_OPEN_EVENT_NAMESPACE);
```

File: js/ui/file_manager/ui.file_manager.item_list.js [hunk 12]
```diff
@@ -1,13 +1,45 @@
 import { extend } from "../../core/utils/extend";
+import dblclickEvent from "../../events/double_click";
```
Comment (DokaRus) [inline js/ui/file_manager/ui.file_manager.item_list.js]:
```suggestion
import { name as dblClickName } from "../../events/double_click";
```

File: js/ui/file_manager/ui.file_manager.item_list.js [hunk 13]
```diff
@@ -1,13 +1,45 @@
...
+        });
+        this._contextMenu.show();
+    }
+
+    _ensureContextMenu() {
+        if(this._contextMenu) {
+            return;
+        }
+
+        this._contextMenu = this._createComponent("<div>", ContextMenu, {
```
Comment (DokaRus) [inline js/ui/file_manager/ui.file_manager.item_list.js]: It would better to render widget at the element attached to DOM

Comment (maxerokh) [reply js/ui/file_manager/ui.file_manager.item_list.js]: Should I use the `this._createComponent($("<div>", ...)` construction?

Comment (DokaRus) [reply js/ui/file_manager/ui.file_manager.item_list.js]: You can make it like below

```js
const $menu = $("<div>").appendTo(this.$element());
this._contextMenu = this._createComponent($menu, ContextMenu, {
```

Comment (vladkovl) [reply js/ui/file_manager/ui.file_manager.item_list.js]: Changed to this version

**Review by DokaRus — APPROVED** [review]

**Pull request merged**